### PR TITLE
feat: JSON Reporter

### PR DESCRIPTION
**Pull request by thejuan**
Adds the ability to define (in config) reporters. Adds a JSON reporter to write lint results to disk in JSON format.

Config:
```
reporters {
    json {
      path = "/tmp/pint.report.json"
    }
  }
```

### CODE REVIEW

File: cmd/pint/lint.go [hunk 1]
```diff
@@ -114,6 +119,20 @@ func actionLint(c *cli.Context) error {
 	return nil
 }
 
+func report(summary reporter.Summary, reporters *config.Reporters) error {
+	if reporters != nil {
```
Comment (mohakkataria) [inline cmd/pint/lint.go]:
```suggestion
	if reporters != nil && reporters.JSON != nil {
```

File: cmd/pint/lint.go [hunk 2]
```diff
@@ -114,6 +119,20 @@ func actionLint(c *cli.Context) error {
...
+		if reporters.JSON != nil {
+			r := reporter.NewJSONReporter(reporters.JSON.Path)
+			err := r.Submit(summary.Reports())
```
Comment (mohakkataria) [inline cmd/pint/lint.go]:
```suggestion
			if err := r.Submit(summary.Reports()); err != nil { return err; }
```
You can shorthand this too like this^

File: internal/config/json_reporter.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package config
+
+import "errors"
+
+type JSONReporterSettings struct {
+	Path string `hcl:"path" json:"path"`
+}
+
+func (ag JSONReporterSettings) validate() error {
```
Comment (mohakkataria) [inline internal/config/json_reporter.go]: Why is the variable named ag? 🤔

File: internal/config/json_reporter_test.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		{
+			conf: JSONReporterSettings{},
+			err:  errors.New("empty path"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("%v", tc.conf), func(t *testing.T) {
+			err := tc.conf.validate()
+			if err == nil || tc.err == nil {
```
Comment (mohakkataria) [inline internal/config/json_reporter_test.go:30]: Ideally this should be
```if tc.err == nil {require.NoError(err) //or similar}```

Comment (thejuan) [reply internal/config/json_reporter_test.go:30]: This is just copy/paste from another testcase. Happy to keep their style if its not major issue?

File: internal/config/json_reporter.go (hunk 3 above)
Comment (thejuan) [reply internal/config/json_reporter.go]: I thought you had to use obscure variable naming in Go? :p
Copy/Paste coding, fixed.

File: internal/config/json_reporter_test.go (hunk 4 above)
Comment (mohakkataria) [reply internal/config/json_reporter_test.go:30]: Sure

**Review by mohakkataria — APPROVED** [review]

**Pull request merged**